extendeddatacrossword.go

```diff
@@ -177,6 +174,10 @@ func (eds *ExtendedDataSquare) solveCrosswordRow(
 				}
 				return false, false, err
 			}
+
+			if eds.verifyEncoding(col, r, rebuiltShares[c]) != nil {
```

Comment:
Something worth consideration is the the precedence of these returned errors i.e., 1) root mismatch (due to the call to `verifyAgainstColRoots`) and 2) the encoding error (caused by `verifyEncoding`). I think if the encoding is already wrong, perhaps it does not matter whether the roots match or not, so returning the encoding verification error would be more accurate and informative. Depending on this decision, we may want to swap the order of the returned errors.  This can be tackled in a separate issue though if further discussion is needed.